rust/src/lib.rs

```diff
@@ -1780,6 +1811,38 @@ enum NativeScriptEnum {
     TimelockExpiry(TimelockExpiry),
 }
 
+#[derive(Debug, Clone, Eq, Ord, PartialEq, PartialOrd)]
+pub enum ScriptRefEnum {
+    NativeScript(NativeScript),
+    PlutusScriptV1(PlutusScript),
+    PlutusScriptV2(PlutusScript),
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct ScriptRef(ScriptRefEnum);
+
+to_from_bytes!(ScriptRef);
+
+#[wasm_bindgen]
+impl ScriptRef {
+
+    pub fn new_native_script(native_script: &NativeScript) -> Self {
+        Self(ScriptRefEnum::NativeScript(native_script.clone()))
+    }
+
+    pub fn new_plutus_script(plutus_script: &PlutusScript) -> Self {
+        //TODO add choose right plutus script after updating type
```

Comment:
After https://github.com/Emurgo/cardano-serialization-lib/pull/450/files#r895454845

Update this to just store the `PlutusScript` in the enum, the script version will be solved on the script level, the ref should be independent of that